flows/runs/run.go

```diff
@@ -102,41 +100,43 @@ func (r *flowRun) SetContact(contact *flows.Contact) { r.contact = contact }
 func (r *flowRun) Context() utils.VariableResolver { return r.context }
 func (r *flowRun) Results() *flows.Results         { return r.results }
 
-func (r *flowRun) IsComplete() bool {
-	return r.status != flows.StatusActive
-}
-func (r *flowRun) setStatus(status flows.RunStatus) {
-	r.status = status
-}
 func (r *flowRun) Exit(status flows.RunStatus) {
-	r.setStatus(status)
+	r.SetStatus(status)
 	now := time.Now().UTC()
 	r.exitedOn = &now
 }
 func (r *flowRun) Status() flows.RunStatus { return r.status }
+func (r *flowRun) SetStatus(status flows.RunStatus) {
+	r.status = status
+}
 
 func (r *flowRun) Parent() flows.FlowRunReference { return r.parent }
 func (r *flowRun) Child() flows.FlowRunReference  { return r.child }
 
-func (r *flowRun) Wait() flows.Wait        { return r.wait }
-func (r *flowRun) SetWait(wait flows.Wait) { r.wait = wait }
-
 func (r *flowRun) Input() flows.Input         { return r.input }
 func (r *flowRun) SetInput(input flows.Input) { r.input = input }
 
-func (r *flowRun) ApplyEvent(s flows.Step, a flows.Action, e flows.Event) {
-	e.Apply(r)
+func (r *flowRun) ApplyEvent(s flows.Step, action flows.Action, event flows.Event) {
+	event.Apply(r)
 
 	fs := s.(*step)
-	fs.addEvent(e)
+	fs.addEvent(event)
 
-	if !e.FromCaller() {
-		r.Session().LogEvent(s, a, e)
+	if !event.FromCaller() {
+		r.Session().LogEvent(s, action, event)
 	}
+
+	// eventEnvelope, _ := utils.EnvelopeFromTyped(event)
+	// eventJSON, _ := json.Marshal(eventEnvelope)
+	// fmt.Printf("⚡︎ in run %s: %s\n", r.UUID(), string(eventJSON))
```

Comment:
I'm usually against leaving cruft in files but this is just sooo useful for debugging - it logs every event applied to the session.